server/cluster/coordinator.go

```diff
@@ -157,6 +157,36 @@ func (c *coordinator) patrolRegions() {
 	}
 }
 
+func (c *coordinator) collectRegionStats() {
+	defer logutil.LogPanic()
+
+	defer c.wg.Done()
+	ticker := time.NewTicker(collectRegionStatsInterval)
+	defer ticker.Stop()
+	log.Info("coordinator starts collect region stats")
+	var key []byte
+	for {
+		select {
+		case <-ticker.C:
+		case <-c.ctx.Done():
+			log.Info("collect region stats has been stopped")
+			return
+		}
+
+		regions := c.cluster.ScanRegions(key, nil, statsScanRegionLimit)
+		if len(regions) == 0 {
+			// Resets the scan key.
+			key = nil
+			continue
+		}
+
+		c.cluster.updateRegionStats(regions)
```

Comment:
We can create `regionStats` and `labelLevelStats` locally and set them into cluster after all regions are collected. No need to lock `cluster` every batch.

Reply:
We still need to get the store so that the lock cannot be eliminated.